crypto/param_build.c

```diff
@@ -277,25 +277,32 @@ static OSSL_PARAM *param_bld_convert(OSSL_PARAM_BLD *bld, OSSL_PARAM *param,
     return param;
 }
 
-OSSL_PARAM *ossl_param_bld_to_param(OSSL_PARAM_BLD *bld, void **secure)
+OSSL_PARAM *ossl_param_bld_to_param(OSSL_PARAM_BLD *bld, void **secure,
+                                    size_t *sec_sz)
 {
     OSSL_PARAM_BLD_BLOCK *blk, *s = NULL;
     OSSL_PARAM *param;
     const size_t p_blks = bytes_to_blocks((bld->curr + 1) * sizeof(*param));
     const size_t total = ALIGN_SIZE * (p_blks + bld->total_blocks);
+    size_t ss;
```

Comment:
For code elegance, the size could be calculated here in the same manner as the other two sizes.

Reply:
Costing a multiply (okay probably a shift) when it isn't required.

Reply:
That's only "costly" if `blk->secure_blocks == 0`, and I wouldn't be surprised if multiplication with zero isn't very costly...

Reply:
You're worried about the cost of a multiple?  Really? Compared to all the effort of the humans to understand, audit, comprehend, etc., the code?